src/agl/wam_ipc.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+package wam_ipc;
+
+service WebAppManagerService {
+  rpc Launch(LaunchRequest) returns (google.protobuf.Empty) {}
+  rpc Activate(ActivateRequest) returns (google.protobuf.Empty) {}
+  rpc Kill(KillRequest) returns (google.protobuf.Empty) {}
+}
+
+message LaunchRequest {
+  string app_id = 1;
+  string uri = 2;
+  int64 surface_id = 3;
+
+  enum SurfaceType {
+    NONE = 0;
+    DESKTOP = 1;
+    BACKGROUND = 2;
+    PANEL = 3;
+    POPUP = 4;
+    FULLSCREEN = 5;
+    SPLIT_V = 6;
+    SPLIT_H = 7;
+    REMOTE = 8;
+  }
```

Comment:
Kind of want to kill some of the roles (they're not really surface types) and incorporate them under the main protocol.

I think it would probably be inconvenient at this point, much more beneficial a later point in time to split these correctly as surface roles and the rest as surface types.

The roles are: DESKTOP/POPUP/FULLSCREEN/SPLIT and finally REMOVE.
The types are BACKGROUND and PANEL. 

The latter are for the agl-shell, the former are from the agl-shell-desktop protocol. I kind want to kill  the last protocol extension incorporate some of these roles into agl-shell, and see what can be directly by just using xdg-shell protocol. The agl-shell protocol defines the shell client, while the other roles are help basically helpers which do not require a bind for the agl-shell protocol interface. 

Avoiding embedding them into a rather big SurfaceType would mean we just need to cut of just that, rather than cherry-picking/modifying all potential callsites or other particularity. 

IDK, I guess that for you decide, how you'd want to structure this part, but at some point this will need some adjustment.

Reply:
Right now, the only surface roles that have specific actions are background and panel. All the other types are ignored. WAM and chromium are not doing anything with agl-shell-desktop protocol yet.

But changing this properly implies changing appinfo.json, so we have different fields for surface_role and surface_type. Let's do this at a later stage. I reported https://jira.automotivelinux.org/browse/SPEC-4505 for that purpose.

Reply:
That looks good to me. Thank you!